cmake/BLTPrivateMacros.cmake

```diff
@@ -842,3 +842,156 @@ macro(blt_clean_target)
+        endif()
+
+        ## Print all such properties that have been SET
+        foreach (prop ${_property_list})
+            string(REPLACE "<CONFIG>" "${CMAKE_BUILD_TYPE}" prop ${prop})
+            get_property(_propval TARGET ${arg_TARGET} PROPERTY ${prop} SET)
+            if ("${_propval}" AND "${prop}" MATCHES "${arg_PROPERTY_NAME_REGEX}" AND "${_propval}" MATCHES "${arg_PROPERTY_VALUE_REGEX}")
+                get_target_property(_propval ${arg_TARGET} ${prop})
+                message (STATUS "[${arg_TARGET} property] ${prop}: ${_propval}")
+            endif()
+        endforeach()
+        unset(_property_list)
+        unset(_propval)
+    endif()
+
+    ## Additionally, output variables generated via blt_register_target of the form "_BLT_<target>_*"
+    if(_is_blt_registered_target)
+        set(_target_prefix "_BLT_${_target_upper}_")
+
+        ## Filter to get variables of the form _BLT_<target>_ and print
+        get_cmake_property(_variable_names VARIABLES)
+        foreach (prop ${_variable_names})
+            if("${prop}" MATCHES "^${_target_prefix}" AND "${prop}" MATCHES "${arg_PROPERTY_NAME_REGEX}" AND "${_propval}" MATCHES "${arg_PROPERTY_VALUE_REGEX}")
+                message (STATUS "[${arg_TARGET} property] ${prop}: ${${prop}}")
+            endif()
+        endforeach()
+        unset(_target_prefix)
+        unset(_variable_names)
+    endif()
+
+    unset(_target_upper)
+    unset(_is_blt_registered_target)
+    unset(_is_cmake_target)
+endmacro(blt_print_target_properties_private)
+
+##------------------------------------------------------------------------------
+## blt_find_all_targets_recursive(TARGET <target> TLIST tlist)
+##
+## Store all target's dependancies (link libraries and interface link libraries)
+## recursively in TLIST.
+##
+##------------------------------------------------------------------------------
+macro(blt_find_all_targets_recursive)
+
+    set(options)
+    set(singleValuedArgs TARGET TLIST)
+    set(multiValuedArgs)
+
+    # parse the arguments to the macro
+    cmake_parse_arguments(arg
+         "${options}" "${singleValuedArgs}" "${multiValuedArgs}" ${ARGN})
+
+    # check for required arguments
+    if(NOT DEFINED arg_TARGET)
+        message(FATAL_ERROR "TARGET is a required parameter for the blt_find_all_targets_recursive macro")
+    endif()
+
+    if(NOT DEFINED ${arg_TLIST})
+        message(FATAL_ERROR "TLIST is a required parameter for the blt_find_all_targets_recursive macro")
+    endif()
+
+    # check if this is a blt_registered target
+    string(TOUPPER ${arg_TARGET} _target_upper)
+    if(_BLT_${_target_upper}_IS_REGISTERED_LIBRARY)
+        # recursive call
+        set (_depends_on "${_BLT_${_target_upper}_DEPENDS_ON}")
+        foreach(t ${_depends_on})
+            list(APPEND ${arg_TLIST} ${t})
+            blt_find_all_targets_recursive(TARGET ${t} TLIST ${arg_TLIST})
+        endforeach()
+        unset(_depends_on)
+    endif()
```

Comment:
Should we be concerned about 
(a) infinite recursion, e.g. if there is a cycle in the dependencies, or 
(b) repeated dependencies -- e.g. we have a DAG of dependencies, and this function, as written will find the dependents of each of the dependents

It might be better to only call `blt_find_all_targets_recursive` once for each potential dependent, e.g. by checking against the list of targets that we've already checked.

Reply:
After `blt_find_target_dependencies` is called, `blt_list_remove_duplicates` is called on the list (in the public `blt_print_target_properties`), so (b) should not be an issue. I'll test infinite recursion

Reply:
Yes when I do

```
blt_register_library(NAME foo DEPENDS_ON bar)
blt_register_library(NAME bar DEPENDS_ON foo)
blt_print_target_properties(TARGET bar CHILDREN true)
```

There is an infinite loop. I can try an solve this issue

Reply:
I now check if a dependency is already in the list before adding it to `TLIST`.